src/model_signing/api.py

```diff
@@ -0,0 +1,582 @@
+        shard_size: int = 1000000,
+    ) -> Callable[[pathlib.Path, int, int], file.ShardedFileHasher]:
+        """Builds the hasher factory for a serialization by file shards.
+
+        Args:
+            hashing_algorithm: the hashing algorithm to use to hash a file
+            chunk_size: The amount of file to read at once. Default is 8KB. A
+              special value of 0 signals to attempt to read everything in a
+              single call.
+            shard_size: The size of a file shard. Default is 1,000,000 bytes.
+
+        Returns:
+            The hasher factory that should be used by the active serialization
+            method.
+        """
+        algorithm = self._build_stream_hasher(hashing_algorithm)
+
+        def factory(
+            path: pathlib.Path, start: int, end: int
+        ) -> file.ShardedFileHasher:
+            return file.ShardedFileHasher(
+                path,
+                algorithm,
+                start=start,
+                end=end,
+                chunk_size=chunk_size,
+                shard_size=shard_size,
+            )
+
+        return factory
+
+    def set_serialize_by_file_to_manifest(
+        self,
+        *,
+        hashing_algorithm: Literal["sha256", "blake2"] = "sha256",
+        chunk_size: int = 8192,
+        max_workers: int | None = None,
+        allow_symlinks: bool = False,
+    ) -> Self:
+        """Configures serialization to a manifest pairing files with hashes.
+
+        The serialization method in this configuration is changed to one where
+        every file in the model is paired with its digest and a manifest
+        containing all these pairings is being returned.
+
+        Args:
+            hashing_algorithm: the hashing algorithm to use to hash a file
+            chunk_size: The amount of file to read at once. Default is 8KB. A
+              special value of 0 signals to attempt to read everything in a
+              single call.
+            max_workers: Maximum number of workers to use in parallel. Default
+              is to defer to the `concurrent.futures` library.
+            allow_symlinks: Controls whether symbolic links are included. If a
+              symlink is present but the flag is `False` (default) the
+              serialization would raise an error.
+
+        Returns:
+            The new hashing configuration with the new serialization method.
+        """
+        self._serializer = serialize_by_file.ManifestSerializer(
+            self._build_file_hasher_factory(hashing_algorithm, chunk_size),
+            max_workers=max_workers,
+            allow_symlinks=allow_symlinks,
+        )
+        return self
+
+    def set_serialize_by_file_to_digest(
+        self,
+        *,
+        hashing_algorithm: Literal["sha256", "blake2"] = "sha256",
+        merge_algorithm: Literal["sha256", "blake2"] = "sha256",
+        chunk_size: int = 8192,
+        allow_symlinks: bool = False,
+    ) -> Self:
```

Comment:
(technically this doesn't match the hashing constructor, but this is also a different manifest type)

Reply:
This is a repeat of the other symlink comments, phrased differently, and can be resolved.

I had only mentioned the different manifest types because it made the difference in their behavior slightly less contradictory.